src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, CompletableFuture<UserData>> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        // Use computeIfAbsent to initiate the data-fetching CompletableFuture only if absent
+        return cache.computeIfAbsent(userId, id -> fetchUserDataAsync(id));
+    }
+
+    // Asynchronous method to simulate fetching data with delay
+    private CompletableFuture<UserData> fetchUserDataAsync(String userId) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                // Simulate a delay for fetching user data
+                TimeUnit.SECONDS.sleep(1); // Simulate delay without blocking the cache
```

Comment:
Simulating a delay with `TimeUnit.SECONDS.sleep(1)` is a blocking operation, which is generally not recommended within a `CompletableFuture`'s lambda. If this is for simulating a database call or an HTTP request, consider using a non-blocking call or a proper async mechanism.